chore(cardinal): remove unused structs in CQL, define component getter function as type

technicallyty opened the pull request:
## Overview

defines the GetComponentByName dependency explicitly, so we dont type out the function signature multiple times

removes unused structs

## Brief Changelog

<!---
Example:
- The metadata is stored in the blob store on job creation time as a persistent artifact
- Deployments RPC transmits only the blob storage reference
- Daemons retrieve the RPC data from the blob cache
--->

## Testing and Verifying

<!---
Pick one of the following options:

- This change is a trivial rework/code cleanup without any test coverage.

- This change is already covered by existing tests, such as <describe test>.

- This change added tests and can be verified as follows:
    - Added unit test that validates ...
    - Added integration tests for end-to-end deployment with ...
    - Extended integration test for ...
    - Manually verified the change by ...
--->

pyrofolium reviewed (approved):
lgtm. Pretty straightforward.

smsunarto commented:
I rebased this to main, update golangci to the new cql.go directory to continue ignoring it, and made one minor change:

Tyler's diff pointed the component getter was looking for "ComponentMetadata" when it should have been "Component". I fixed this accordingly.

technicallyty commented on cardinal/cql/cql.go:
```diff
@@ -248,12 +240,10 @@ func termToComponentFilter(
-) (filter.ComponentFilter, error) {
-	term, err := internalCQLParser.ParseString("", cqlText)
+func Parse(cqlText string, stringToComponent componentByName) (filter.ComponentFilter, error) {
+	term, err := internalCQLParser.ParseString("failed to parse CQL string", cqlText)
 	if err != nil {
-		return nil, eris.Wrap(err, "failed to parse CQL string")
+		return nil, eris.Wrap(err, "")
```
ser @smsunarto 👁️ 👄 👁️

smsunarto replied:
whoops sorry

technicallyty commented:
> I rebased this to main, update golangci to the new cql.go directory to continue ignoring it, and made one minor change:
> 
> Tyler's diff pointed the component getter was looking for "ComponentMetadata" when it should have been "Component". I fixed this accordingly. CQL doesn't need ComponentMetadata; Component is more than sufficient. ComponentMetadata creates tight coupling because it is intended for World internals.
> 
> If this is acceptable, should be good to merge the PR.

fine with me. Is there a plan to make metadata private? They both come from the same package, so am confused on the coupling part.

technicallyty commented:
> > > I rebased this to main, update golangci to the new cql.go directory to continue ignoring it, and made one minor change:
> > > Tyler's diff pointed the component getter was looking for "ComponentMetadata" when it should have been "Component". I fixed this accordingly. CQL doesn't need ComponentMetadata; Component is more than sufficient. ComponentMetadata creates tight coupling because it is intended for World internals.
> > > If this is acceptable, should be good to merge the PR.
> > 
> > 
> > fine with me. Is there a plan to make ComponentMetadata private? They both come from the same package, so am confused on the coupling part.
> 
> ComponentMetadata is now moved to `cardinal/component` via my PR; this will create more clarity that this is an internal-facing type by moving it out of the user focused `cardinal` package.

ahhhh okay that makes sense. thx for clarifying🤠
